keps/sig-apps/20190226-maxunavailable-for-statefulsets.md

```diff
@@ -125,21 +126,44 @@ type RollingUpdateStatefulSetStrategy struct {
 - If a partition is specified, maxUnavailable will only apply to all the pods which are staged by the partition. Which means all Pods 
   with an ordinal that is greater than or equal to the partition will be updated when the StatefulSet’s .spec.template is updated. Lets 
   say total replicas is 5 and partition is set to 2 and maxUnavailable is set to 2. If the image is changed in this scenario, following
-  are the behavior choices we have:-
-  -  pods with ordinal 4 and 3 will go down at the same time(because of maxUnavailable). Once they are both running and ready, pods with 
-     ordinal 2 will go down. Pods with ordinal 0 and 1 will remain untouched due the partition.
-  -  pods with ordinal 4 and 3 will go down at the same time(because of maxUnavailable). When any of 4 or 3 are running and ready, pods
-     with ordinal 2 will start going down. This could violate ordering guarantees, since if 3 is running and ready, then both 4 and 2
-     are terminating at the same time out of order.
-  -  pod with ordinal 4 and 3 will go down at the same time(because of maxUnavailable). When 4 is running and ready, 2 will go down. At
-     this time both 2 and 3 are terminating. If 3 is running and ready before 4, 2 wont go down to preserve ordering semantics. So at 
-     this time, only 1 is unavailable although we requested 2.
+  are the possible behavior choices we have:-
+
+  1: Pods with ordinal 4 and 3 will start Terminating at the same time(because of maxUnavailable). Once they are both running and ready, pods 
+     with ordinal 2 will start Terminating. Pods with ordinal 0 and 1 will remain untouched due the partition. In this choice, the number of pods
+     terminating is not always maxUnavailable, but sometimes less than that. For e.g. if pod with ordinal 3 is running and ready but 4 is not, we
+     still wait for 4 to be running and ready before moving on to 2. This implementation avoids out of order Terminations of pods.
+  2: Pods with ordinal 4 and 3 will start Terminating at the same time(because of maxUnavailable). When any of 4 or 3 are running and ready, pods
```

Comment:
The three alternatives are getting merged into one paragraph; needs an extra newline between each alternative.

Reply:
add a newline